File: PHP/date.php
<?php

class Date {
    public static function getLocalTime($timezone) {
        $now = time();
        $now += $timezone * 3600;
        return $now;
    }

    public static function now() {
        $d = new Date();
        return $d . " " . Date::clock();
    }

    public static function clock() {
        $now = Date::getLocalTime((new Date())->month < 7 ? 4.5 : 3.5);
        $second = $now % 60;
        $minute = (int)($now / 60) % 60;
        $hour = (int)($now / 3600) % 24;
        return $hour . ":" . $minute . ":" . $second;
    }

    private $year;
    private $month;
    private $day;
    private $dayOfWeek;

    private function calcDayOfWeek() {
        $monthDays = ($this->month - 1) * 31 - ($this->month < 8 ? 0 : $this->month - 7);
        $this->dayOfWeek = ($this->year . $this->year / 4 . $this->day . $monthDays) % 7;
    }

    private function setToNow($timezone) {
        $now = Date::getLocalTime($timezone);
        $now = $now / 3600;
        $now = $now / 24;
        $dif = (int)($now % (365 * 4 + 1));
        $this->year = (int)($now / (365 * 4 + 1)) * 4 + 1348;
        $this->day = 11;
        $this->month = 10;
        for ($i = 0; $i < $dif; $i++) {
            $this->goNext();
        }
        $this->dayOfWeek = (int)((5 + $now) % 7);
    }

    private function isItKabise() {
        $arr = [1, 5, 9, 13, 17, 22, 26, 30];
        return in_array($this->year % 33, $arr);
    }

    public function __construct($y = "", $m = "", $d = "") {
        if ($y == "") {
            $this->setToNow(3.5);
            if ($this->month < 7) $this->setToNow(4.5);
        } else if ($m == "") {
            $tokens = explode("/", $y);
            if (count($tokens) != 3) throw new Exception("Date string is invalid");
            $this->year = (int)($tokens[0]);
            $this->month = (int)($tokens[1]);
            $this->day = (int)($tokens[2]);
            $this->calcDayOfWeek();
        } else {
            $this->year = $y;
            $this->month = $m;
            $this->day = $d;
            $this->calcDayOfWeek();
        }
    }

    public function goNext() {
        $this->day++;
        if ($this->month < 7) {
            if ($this->day > 31) {
                $this->day = 1;
                $this->month++;
            }
        } else {
            if ($this->month != 12 || $this->isItKabise()) {
                if ($this->day > 30) {
                    $this->day = 1;
                    $this->month++;
                }
            } else {
                if ($this->day > 29) {
                    $this->day = 1;
                    $this->month++;
                }
            }
        }
        if ($this->month > 12) {
            $this->month = 1;
            $this->year++;
        }
        $this->dayOfWeek = ($this->dayOfWeek + 1) % 7;
    }

    public function goPrev() {
        $this->day--;
        if ($this->day == 0) {
            $this->day = 31;
            $this->month--;
            if ($this->month == 0) {
                $this->month = 12;
                $this->year--;
            }
            if ($this->month > 6) {
                $this->day--;
            }
            if ($this->month == 12 && !$this->isItKabise()) {
                $this->day--;
            }
            $this->dayOfWeek = ($this->dayOfWeek - 1) % 7;
        }
    }

    public function prev() {
        $d = clone $this;
        $d->goPrev();
        return $d;
    }

    public function next() {
        $d = clone $this;
        $d->goNext();
        return $d;
    }

    public function __toString() {
        return $this->year . "/" . $this->month . "/" . $this->day;
    }

    public function getDayOfWeek() {
        switch ($this->dayOfWeek) {
            case 0:
                return "شنبه";
            case 1:
                return "یک شنبه";
            case 2:
                return "دوشنبه";
            case 3:
                return "سه شنبه";
            case 4:
                return "چهارشنبه";
            case 5:
                return "پنج شنبه";
            case 6:
                return "جمعه";
        }
        throw new Exception("DayOfWeek is wrong");
    }
}

File: C# WPF/Date.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MPD
{
    class Date
    {
        public static long getLocalTime(double timezone)
        {
            long now=(long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
            now += (long)(timezone * 3600);
            return now;
        }
        public static string now()
        {
            Date d=new Date();
            return d.ToString() + " " + clock();
        }
        public static string clock()
        {
            long now = getLocalTime(new Date().month < 7 ? 4.5 : 3.5);
            long second, minute, hour;
            second = now % 60;
            minute = (now / 60) % 60;
            hour = (now / 3600) % 24;
            return hour + ":" + minute + ":" + second;
        }

        private int year;
        private int month;
        private int day;
        private int dayOfWeek;
        private Date clone()
        {
            Date d = new Date();
            d.dayOfWeek = dayOfWeek;
            d.day = day;
            d.month = month;
            d.year = year;
            return d;
        }
        private void calcDayOfWeek()
        {
            int monthDays = (month - 1) * 31 - (month < 8 ? 0 : month - 7);
            dayOfWeek = (year + year / 4 + day + monthDays) % 7;
        }
        private void setToNow(double timezone)
        {
            long now = getLocalTime(timezone);
            int dif = 0;
            now = now / 3600;
            now = now / 24;
            dif = (int)(now % (365 * 4 + 1));
            year = (int)(now / (365 * 4 + 1)) * 4 + 1348;
            day = 11;
            month = 10;
            for (int i = 0; i < dif; i++)
            {
                goNext();
            }
            dayOfWeek = (int)((5 + now) % 7);
        }
        private bool isItKabise()
        {
            int[] arr = { 1, 5, 9, 13, 17, 22, 26, 30 };
            return Array.Exists<int>(arr, i => i == year % 33);
        }
        public Date()
        {
            setToNow(3.5);
            if (month < 7) setToNow(4.5);
        }
        public Date(int y, int m, int d)
        {
            year = y;
            month = m;
            day = d;
            calcDayOfWeek();
        }
        public Date(string input)
        {
            String[] tokens = input.Split('/');
            if (tokens.Length != 3) throw new Exception("Date string is invalid");
            year = Convert.ToInt32(tokens[0]);
            month = Convert.ToInt32(tokens[1]);
            day = Convert.ToInt32(tokens[2]);
            calcDayOfWeek();
        }
        public void goNext()
        {
            day++;
            if (month < 7)
            {
                if (day > 31)
                {
                    day = 1;
                    month++;
                }
            }
            else
            {
                if (month != 12 || isItKabise())
                {
                    if (day > 30)
                    {
                        day = 1;
                        month++;
                    }
                }
                else
                {
                    if (day > 29)
                    {
                        day = 1;
                        month++;
                    }
                }
            }
            if (month > 12)
            {
                month = 1;
                year++;
            }
            dayOfWeek = (dayOfWeek + 1) % 7;
        }
        public void goPrev()
        {
            day--;
            if (day == 0)
            {
                day = 31;
                month--;
                if (month == 0)
                {
                    month = 12;
                    year--;
                }
                if (month > 6)
                {
                    day--;
                }
                if (month == 12 && !isItKabise())
                {
                    day--;
                }
                dayOfWeek = (dayOfWeek - 1) % 7;
            }
        }
        public Date prev()
        {
            Date d = clone();
            d.goPrev();
            return d;
        }
        public Date next()
        {
            Date d = clone();
            d.goNext();
            return d;
        }
        public override string ToString()
        {
            return year + "/" + month + "/" + day;
        }
        public string getDayOfWeek()
        {
            switch (dayOfWeek)
            {
                case 0:
                    return "شنبه";
                case 1:
                    return "یک شنبه";
                case 2:
                    return "دوشنبه";
                case 3:
                    return "سه شنبه";
                case 4:
                    return "چهارشنبه";
                case 5:
                    return "پنج شنبه";
                case 6:
                    return "جمعه";
            }
            throw new Exception("DayOfWeek is wrong");
        }
    }
}
